### PR TITLE
Add table gcp_project.Closes #199

### LINKED ISSUE #199: Add table gcp_project
**References**
https://cloud.google.com/resource-manager/reference/rest/v1/projects/get

### PR DESCRIPTION
**Pull request by anisadas**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
SETUP: tests/gcp_project []

PRETEST: tests/gcp_project

TEST: tests/gcp_project
Running terraform
data.google_client_config.current: Refreshing state...
data.google_project.current: Refreshing state...
data.null_data_source.resource: Refreshing state...

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

current_project_name = parker-aaa
project_id = parker-aaa
resource_aka = gcp://cloudresourcemanager.googleapis.com/projects/parker-aaa

Running SQL query: test-list-query.sql
[
  {
    "name": "parker-aaa",
    "project_id": "parker-aaa"
  }
]
✔ PASSED

Running SQL query: test-notfound-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "akas": [
      "gcp://cloudresourcemanager.googleapis.com/projects/parker-aaa"
    ],
    "name": "parker-aaa",
    "title": "parker-aaa"
  }
]
✔ PASSED

POSTTEST: tests/gcp_project

TEARDOWN: tests/gcp_project

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
select
  name,
  project_id,
  project_number,
  lifecycle_state,
  create_time
from
  gcp_project;
+------------+------------+----------------+-----------------+---------------------+
| name       | project_id | project_number | lifecycle_state | create_time         |
+------------+------------+----------------+-----------------+---------------------+
| parker-aaa | parker-aaa | 979620418102   | ACTIVE          | 2019-05-22 12:08:03 |
+------------+------------+----------------+-----------------+---------------------+


```
</details>

### CODE REVIEW

File: gcp/table_gcp_project.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromConstant("global"),
+			},
+			{
+				Name:        "project",
+				Description: ColumnDescriptionProject,
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getProject,
+				Transform:   transform.FromValue(),
+			},
```
Comment (Subhajit97) [inline gcp/table_gcp_project.go:79]: Not required

File: gcp/table_gcp_project.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+			{
+				Name:        "labels",
+				Description: "A list of labels attached to this project.",
+				Type:        proto.ColumnType_JSON,
+			},
+
+			// standard steampipe columns
+			{
+				Name:        "akas",
```
Comment (Subhajit97) [inline gcp/table_gcp_project.go:59]: Add `title` column

File: gcp/table_gcp_project.go [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+	return nil, nil
+}
+
+func getProjectAka(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	// Get project details
+	projectData, err := activeProject(ctx, d)
+	if err != nil {
+		return nil, err
+	}
+	project := projectData.Project
```
Comment (Subhajit97) [inline gcp/table_gcp_project.go:115]: Do we need this block?
As we are getting the project in response.

File: gcp/table_gcp_project.go [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+			},
+			{
+				Name:        "project_id",
+				Description: "The unique, user-assigned ID of the Project.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "project_number",
+				Description: "The number uniquely identifying the project.",
+				Type:        proto.ColumnType_STRING,
```
Comment (Subhajit97) [inline gcp/table_gcp_project.go:34]: Can we convert the value in `int64` format?

File: docs/tables/gcp_project.md [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+## Examples
+
+### Basic info
+
+```sql
+select
+  *
+from
+  gcp_project;
```
Comment (Subhajit97) [inline docs/tables/gcp_project.md:13]: Can we query specific columns instead of `*`?

**Review by Subhajit97 — CHANGES_REQUESTED** [review]
@anisadas There are few changes required. Please see comments, thanks!
Can you also add the integration test for the table?

File: gcp/table_gcp_project.go (hunk 3 above)
Comment (anisadas) [reply gcp/table_gcp_project.go:115]: This is required  to filter out the current project in list call

Comment (Subhajit97) [reply gcp/table_gcp_project.go:106]: @anisadas We are filtering project in GET call.
Why we need to filter when building `akas`?

Comment (anisadas) [reply gcp/table_gcp_project.go:106]: My bad I think you are talking this  block in list call . I understood it. Will change it

File: gcp-test/tests/gcp_project/test-list-expected.json [hunk 6]
```diff
@@ -0,0 +1,6 @@
+[
+  {
+    "name": "{{ output.current_project_name.value }}",
+		"project_id": "{{ output.project_id.value }}"
```
Comment (Subhajit97) [inline gcp-test/tests/gcp_project/test-list-expected.json:4]: Fix indentation

File: gcp-test/tests/gcp_project/variables.tf [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+data "google_client_config" "current" {}
+
+data "null_data_source" "resource" {
+  inputs = {
+    scope = "gcp://cloudresourcemanager.googleapis.com/projects/${data.google_client_config.current.project}"
+  }
+}
+
+data "google_project" "current" {
+}
```
Comment (Subhajit97) [inline gcp-test/tests/gcp_project/variables.tf:40]:
```suggestion
data "google_project" "current" {}
```

File: gcp-test/tests/gcp_project/variables.tf [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+
+output "resource_aka" {
+  value = data.null_data_source.resource.inputs.scope
+}
+
```
Comment (Subhajit97) [inline gcp-test/tests/gcp_project/variables.tf:45]: Can we also validate `project_number`?
```
output "project_number" {
  value = data.google_project.current.number
}
```

File: gcp/table_gcp_project.go [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+		return nil, err
+	}
+
+	return nil, nil
+}
+
+func getProjectAka(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	// Get project details
+	project := h.Item.(*cloudresourcemanager.Project)
+	// Build resource aka
```
Comment (Subhajit97) [inline gcp/table_gcp_project.go:104]: Add a new line
```suggestion

	// Build resource aka
```

File: gcp/table_gcp_project.go [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+		},
+		Columns: []*plugin.Column{
+			{
+				Name:        "name",
+				Description: "The name of the project.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "project_id",
+				Description: "The unique, user-assigned ID of the Project.",
```
Comment (Subhajit97) [inline gcp/table_gcp_project.go:29]:
```suggestion
				Description: "An unique, user-assigned ID of the Project.",
```

File: gcp/table_gcp_project.go [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+				Description: "An optional reference to a parent Resource.",
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name:        "labels",
+				Description: "A list of labels attached to this project.",
+				Type:        proto.ColumnType_JSON,
+			},
+
+			// standard steampipe columns
```
Comment (Subhajit97) [inline gcp/table_gcp_project.go:58]:
```suggestion
			// Steampipe standard columns
```

File: gcp/table_gcp_project.go [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+	for _, project := range resp.Projects {
+		d.StreamListItem(ctx, project)
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	return nil, nil
+}
+
```
Comment (Subhajit97) [inline gcp/table_gcp_project.go:100]:
```suggestion

//// HYDRATE FUNCTIONS
```

**Comment by anisadas** [comment]
```
SETUP: tests/gcp_project []

PRETEST: tests/gcp_project

TEST: tests/gcp_project
Running terraform
data.google_client_config.current: Refreshing state...
data.google_project.current: Refreshing state...
data.null_data_source.resource: Refreshing state...

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

current_project_name = parker-aaa
project_id = parker-aaa
project_number = 979620418102
resource_aka = gcp://cloudresourcemanager.googleapis.com/projects/parker-aaa

Running SQL query: test-list-query.sql
[
  {
    "name": "parker-aaa",
    "project_id": "parker-aaa",
    "project_number": 979620418102
  }
]
✔ PASSED

Running SQL query: test-notfound-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "akas": [
      "gcp://cloudresourcemanager.googleapis.com/projects/parker-aaa"
    ],
    "name": "parker-aaa",
    "title": "parker-aaa"
  }
]
✔ PASSED

POSTTEST: tests/gcp_project

TEARDOWN: tests/gcp_project

SUMMARY:

1/1 passed.
```

**Review by Subhajit97 — APPROVED** [review]
LGTM

**Review by rajmohanty17 — APPROVED** [review]
LGTM

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**